source/extensions/common/proxy_protocol/proxy_protocol_header.cc

```diff
@@ -118,6 +127,27 @@ void generateV2LocalHeader(Buffer::Instance& out) {
   out.add(addr_fam_protocol_and_length, 4);
 }
 
+void generateV2Header(const Network::ProxyProtocolData& prox_proto_data, Buffer::Instance& out) {
+  uint16_t extension_length = 0;
+  for (auto& tlv : prox_proto_data.tlv_vector_) {
+    // TODO: handle overflow without crashing.
+    // assert(tlv.value.size() + 3u < std::numeric_limits<uint16_t>::max() - extension_length);
```

Comment:
This line is commented because of type mismatch, but more importantly, I'm not sure of the best way to handle errors at this deep level. Shall I throw an exception, bubble up the result via the returned value, or assert here?

I'd planned to fix the type issue, go with assertion and bring this question to Envoy code reviewers.

Reply:
Thanks for the references and suggestions! I think exception makes the most sense since the TLVs are provided by downstreams.
> Triggerable or uncertain conditions, may be based on untrusted data plane traffic or an extensions’ contract.